### ISSUE TITLE
Socket.dev reports on all our repos

### ISSUE THREAD

**Issue by wesleytodd** [expressjs/security-wg#17]
I think we should add the Socket.dev tooling to all three orgs repos. It is easy to do, I have already done it for some personal repos and some work ones. I find it is the best in class for warning me about changes I might care about in updated dependencies. Anyone opposed to this? I could add it just to one for the time being if we wanted to try it out first.

cc @expressjs/express-tc @expressjs/security-wg

**Comment by ctcpip** [comment]
I don't really see any downside aside from the noise from comments.  (Which shouldn't be just noise if it's doing anything useful.)  And we can always remove it.

**Comment by wesleytodd** [comment]
The one thing I am not sure is how well it reports when we dont have lockfiles. There are good reasons not to have lock files for libraries, but it might be another thing to consider at some point. If we can get reliable automation around testing before publish with fully updated locks then ideally we can add them, but for now even just something to help tell us that a [PR like this](https://github.com/expressjs/body-parser/pull/521) did not pull in anything surprising would be nice.

**Comment by UlisesGascon** [comment]
> I think we should add the Socket.dev tooling to all three orgs repos. 

Yes! 100%

> The one thing I am not sure is how well it reports when we dont have lockfiles.

We can do a test and if this is an issue we can revert the integration or explore alternatives.

**Comment by ctcpip** [comment]
> The one thing I am not sure is how well it reports when we dont have lockfiles.

IME with these tools, the only way to know for sure is to fork a repo and compare the results with and without a lock file.

**Comment by UlisesGascon** [comment]
Seems like we are good to add it. We can do a fast check with the team in the next TC meeting and enable it

**Comment by wesleytodd** [comment]
Yep I will add the agenda label. But yeah then I can enable it on a few to start.

**Comment by bjohansebas** [comment]
I think this is already complete, right? I see it has already been integrated into several repositories that weren't included before

**Comment by wesleytodd** [comment]
It is partially complete, but I think I need to add it in a few more things to complete it. Once I have time to verify it all I will circle back and close this.

**Comment by bjohansebas** [comment]
Also, I can't see the reports generated by socket. Is there a way to make them public?

**Comment by wesleytodd** [comment]
I have that on my list to ask about. Just haven't gotten to that yet.

**Comment by bjohansebas** [comment]
Apparently, it needs to be enabled in Express, he didn't generate the report in this [PR](https://github.com/expressjs/express/pull/6332).

**Comment by wesleytodd** [comment]
Just enabled it. I might need to poke the PR for it to run. I will do that later when I go to merge.

**Comment by bjohansebas** [comment]
It seems like it wasn't reported here either, and this is a new dependency. https://github.com/pillarjs/router/pull/151

**Comment by wesleytodd** [comment]
Bradley from Socket DM'd me about it and said that it looks like the GH api was not responding beyond their normal retries. He is looking into options to increase reliability and maybe add a manual retry command. I think in the mean time we can look for them in the dashboard, for example here is the one for that PR: https://socket.dev/dashboard/org/pillarjs/sbom/78ff2c40-f85f-4da2-9210-93a5ddafb36c?action=error%2Cmonitor%2Cwarn

**Comment by UlisesGascon** [comment]
I think that at this stage we can close this issue as the tool is so far working fine 👍
